### PR TITLE
Provide DBs max string lengths as InitialState

### LINKED ISSUE #92: Increase length for description and show proper errors
**Which Forms version are you running?**
1.0.3

**Nextcloud or ownCloud?:**
Nextcloud

**Nextcloud/ownCloud version:** (see Nextcloud admin page)
16.0.4

### Steps to reproduce
1. Try to submit a form with a description with more than 1024 characters

### Actual behaviour
Form does not get saved and shows a generic "Error on saving form, see console".
Request gives a 500 Internal Server Error.

### Expected behaviour
Form does get saved (or at least an error message saying that descriptions with more than 1024 characters are not supported).

### PR DESCRIPTION
**Pull request by jotoeri**
Closes #92

I gave a try to the inital-state thing. Hope this is the right place to provide the State? Unfortunately the loadState is not working. Browser console tells me `Error: "Could not find initial state maxStringLengths of forms"`

Can you identify why?

### CODE REVIEW

**Comment by skjnldsv** [comment]
> Can you identify why?

The initial state needs to be provided by the page. So not the api controller.

**Comment by jotoeri** [comment]
OK, thank you, @skjnldsv 

Now it works, i think i tested all cases.

File: src/components/Questions/QuestionLong.vue [hunk 1]
```diff
@@ -54,6 +55,7 @@ export default {
 	data() {
 		return {
 			height: 1,
+			maxStringLengths: loadState('forms', 'maxStringLengths'),
```
Comment (skjnldsv) [inline src/components/Questions/QuestionLong.vue:58]: If all of them have the same data, please use the Question mixin so it doesn't duplicate code :)
Btw, I thought we had different length for short and long answers?

Comment (skjnldsv) [reply src/components/Questions/QuestionLong.vue:58]: Ah sorry, we do. I misread. One uses `optionText`, the other `answerText`

Comment (skjnldsv) [reply src/components/Questions/QuestionLong.vue:58]: Still, but those into the Mixin for better code styling ;)

File: lib/Controller/PageController.php [hunk 2]
```diff
@@ -182,6 +204,7 @@ public function gotoForm($hash): ?TemplateResponse {
 
 		Util::addScript($this->appName, 'submit');
 		$this->initialStateService->provideInitialState($this->appName, 'form', $this->formsService->getForm($form->getId()));
+		$this->provideStaticData();
```
Comment (skjnldsv) [inline lib/Controller/PageController.php:207]: I would rather have this directly here.
Not calling an external function :)

Put the maxStringLengths as constantes too, that is what it's here for 

Regarding the number of duplicate functions like editForm, cloneForm... etc etc, we will put them all into one in a followup pr, we can use the same endpoint for multiple routes with a little trick ;)

File: src/components/Questions/QuestionLong.vue (hunk 1 above)
Comment (jotoeri) [reply src/components/Questions/QuestionLong.vue:58]: Ah, yes, will move this to the mixin...

The option & answer is different thing. Options-Table is the one containing options for multiple-questions (1024 chars), while the answer-table is the one containing user-answers (2048 char as it must be able to take all texts, including long-text answers). Even options are stored within the answer-table, when chosen by the user as answer on submit.
For now i just set the short answer-text to the same max-length, as this is the restriction by DB. If we want, we can restrict short answers to a smaller number, DB won't care^^. But i don't know if it's really necessary...

File: lib/Controller/PageController.php (hunk 2 above)
Comment (jotoeri) [reply lib/Controller/PageController.php:207]: Thought we would maybe want to provide more data like this, this was why i chose this way. But yes, for now i can also move it to a constant... 👍

File: src/components/Questions/QuestionLong.vue (hunk 1 above)
Comment (jotoeri) [reply src/components/Questions/QuestionLong.vue]: Moved it now up even one more step, so it gets loaded only once on create/submit.

**Comment by jotoeri** [comment]
:D I'm done with this. Took me too much time already. ;)

Just to understand - A mixin is compiled/executed everywhere, where it is included, no? So it would load the state on each vue again, but wouldn't matter, as this is what the StateService is for?^^

**Pull request merged**